pennylane/beta/tapes/qnode.py

```diff
@@ -0,0 +1,542 @@
+        interface (str): The interface that will be used for classical backpropagation.
+            This affects the types of objects that can be passed to/returned from the QNode:
+
+            * ``interface='autograd'``: Allows autograd to backpropagate
+              through the QNode. The QNode accepts default Python types
+              (floats, ints, lists) as well as NumPy array arguments,
+              and returns NumPy arrays.
+
+            * ``interface='torch'``: Allows PyTorch to backpropogate
+              through the QNode. The QNode accepts and returns Torch tensors.
+
+            * ``interface='tf'``: Allows TensorFlow in eager mode to backpropogate
+              through the QNode. The QNode accepts and returns
+              TensorFlow ``tf.Variable`` and ``tf.tensor`` objects.
+
+            * ``None``: The QNode accepts default Python types
+              (floats, ints, lists) as well as NumPy array arguments,
+              and returns NumPy arrays. It does not connect to any
+              machine learning library automatically for backpropagation.
+
+        diff_method (str, None): the method of differentiation to use in the created QNode
+
+            * ``"best"``: Best available method. Uses classical backpropagation or the
+              device directly to compute the gradient if supported, otherwise will use
+              the analytic parameter-shift rule where possible with finite-difference as a fallback.
+
+            * ``"backprop"``: Use classical backpropagation. Only allowed on simulator
+              devices that are classically end-to-end differentiable, for example
+              :class:`default.tensor.tf <~.DefaultTensorTF>`. Note that the returned
+              QNode can only be used with the machine-learning framework supported
+              by the device; a separate ``interface`` argument should not be passed.
+
+            * ``"reversible"``: Uses a reversible method for computing the gradient.
+              This method is similar to ``"backprop"``, but trades off increased
+              runtime with significantly lower memory usage. Compared to the
+              parameter-shift rule, the reversible method can be faster or slower,
+              depending on the density and location of parametrized gates in a circuit.
+              Only allowed on (simulator) devices with the "reversible" capability,
+              for example :class:`default.qubit <~.DefaultQubit>`.
+
+            * ``"device"``: Queries the device directly for the gradient.
+              Only allowed on devices that provide their own gradient computation.
+
+
+            * ``"parameter-shift"``: Use the analytic parameter-shift
+              rule where possible, with finite-difference as a fallback.
+
+            * ``"finite-diff"``: Uses numerical finite-differences for all parameters.
+
+    Keyword Args:
+        h=1e-7 (float): step size for the finite difference method

```

Comment:
Specifying the default value this way seems new. Would we like to have this convention from now on?

Reply:
That's a good question. I was curious how this would look, and whether it would render correctly. Seems to look okay:

![image](https://user-images.githubusercontent.com/2959003/93541180-d7401a80-f994-11ea-8af7-3346c0677d78.png)

It also avoids having to (sometimes awkwardly) include the default in the description, e.g., 'Default is __ if not specified'.

Reply:
Think it looks cool! :100: :slightly_smiling_face: